sysinfo.ts

```diff
@@ -0,0 +1,99 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import os = require("os");
+import osenv = require("osenv");
+import Future = require("fibers/future");
+import hostInfo = require("./host-info");
+
+export class SysInfo implements ISysInfo {
+	constructor(private $childProcess: IChildProcess) { }
+
+	private sysInfoCache: ISysInfoData = undefined;
+
+	getSysInfo(): ISysInfoData {
+		if (!this.sysInfoCache) {
+			var res: ISysInfoData = Object.create(null);
+
+			var packageJson = require("../../package.json");
+			res.procInfo = packageJson.name + " / " + packageJson.version;
+
+			// os stuff
+			res.platform = os.platform();
+			res.os = hostInfo.isWindows() ? this.winVer() : this.unixVer();
+			res.shell = osenv.shell();
+			res.dotNetVer = hostInfo.isWindows() ? hostInfo.dotNetVersion(".Net is not installed").wait() : undefined;
+
+			// node stuff
+			res.nodeVer = process.version;
+			res.npmVer = this.$childProcess.exec("npm -v").wait().split("\n")[0];
+			res.procArch = process.arch;
+
+			// dependencies
+			try {
+				res.javaVer = this.$childProcess.spawnFromEvent("java", ["-version"], "exit").wait().stderr.split(os.EOL)[2];
+			} catch(e) {
+				res.javaVer = null;
+			} // if we got an error, assume not working
+
+			res.antVer = this.exec("ant -version").split(os.EOL)[0];
```

Comment:
Because the output of ant -version is actually 2 lines, and the second one is empty.